Core/Core/Common/CommonUI/CoreWebView/View/CoreWebView.swift

```diff
@@ -729,3 +740,98 @@ extension CoreWebView {
         }
     }
 }
+
+// MARK: - Attachment Download
+
+public struct CoreWebAttachment {
+    let url: URL
+    let contentType: String?
+
+    fileprivate init(url: URL, contentType: String?) {
+        self.url = url
+        self.contentType = contentType
+    }
+}
+
+extension CoreWebView: WKDownloadDelegate {
+
+    public var isDownloadingAttachment: Bool { downloadingAttachment != nil }
+
+    public func download(_ download: WKDownload, decideDestinationUsing response: URLResponse, suggestedFilename: String) async -> URL? {
+
+        guard response.httpHeaders.hasAttachmentContentDisposition else { return nil }
+
+        let suggestedUrl = URL.Directories.temporary.appending(component: suggestedFilename)
+        let attachment = CoreWebAttachment(
+            url: suggestedUrl,
+            contentType: response.httpHeaders.contentType
+        )
+
+        downloadingAttachment = attachment
+
+        do {
+            if FileManager.default.fileExists(atPath: suggestedUrl.path()) {
+                try FileManager.default.removeItem(at: suggestedUrl)
+            }
+
+            linkDelegate?.coreWebView(self, didStartDownloadAttachment: attachment)
+
+            return suggestedUrl
+        } catch {
+
+            return nil
+        }
+    }
+
+    public func download(_ download: WKDownload, didFailWithError error: any Error, resumeData: Data?) {
+        guard
+            let attachment = downloadingAttachment,
+            let fileURL = download.progress.fileURL,
+            fileURL == attachment.url
+        else { return }
+
+        linkDelegate?.coreWebView(self, didFailAttachmentDownload: attachment, with: error)
+        downloadingAttachment = nil
+    }
+
+    public func downloadDidFinish(_ download: WKDownload) {
+        guard
+            let attachment = downloadingAttachment,
+            let fileURL = download.progress.fileURL,
+            fileURL == attachment.url
+        else { return }
+
+        linkDelegate?.coreWebView(self, didFinishAttachmentDownload: attachment)
+        downloadingAttachment = nil
+    }
+}
+
+// MARK: - Utils
+
+private extension URLResponse {
+
+    struct HTTPHeaders {
+        let response: URLResponse
+
+        var contentDisposition: String? {
+            headerValue(of: "content-disposition")
+        }
+
+        var hasAttachmentContentDisposition: Bool {
+            return contentDisposition.flatMap({ $0.hasPrefix("attachment") }) ?? false
+        }
+
+        var contentType: String? {
+            headerValue(of: "content-type")
+        }
+
+        private func headerValue(of key: String) -> String? {
+            guard let headers = (response as? HTTPURLResponse)?.allHeaderFields else { return nil }
+            return headers
+                .first(where: { ($0.key as? String)?.lowercased() == key.lowercased() })?
+                .value as? String
+        }
+    }
```

Comment:
I suggest to move these directly to the `URLResponseExtensions` file without this `HTTPHeader` struct. We have such lookups in that extension and the `headerValue` method seems to be generic enough to be used elsewhere. Also we could use the `HTTPHeaderConstants` file if we have such.